### PR TITLE
FISH-9002 EE11 Authentication TCK Runner

### PR DESCRIPTION
**Pull request by Pandrex247**
Updates the TCK runner for EE11.

The `Profile SPI Servlet 3.1.0` and `Profile SPI SOAP 3.1.0` modules still currently fail, and I'm a little unsure if that's being caused by runner config or Payara.

The Servlet tests fail because they seem to have issues writing to the `authentication-trace-log.xml` file - nothing gets written to it and then the test throws these errors when trying to parse it (as it's an empty file):
```
org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 2; The markup in the document preceding the root element must be well-formed.
        at java.xml/com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:262)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:342)
        at java.xml/javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:122)
        at ee.jakarta.tck.authentication.test.common.logging.client.LogFileProcessor.fetchLogs(LogFileProcessor.java:144)
```
The server logs for this test show no obvious errors.

The SOAP tests do manage to correctly write to this `authentication-trace-log.xml` file, but fail with various authentication and `AuthConfigFactory.getConfigProvider` exceptions.

```
[ERROR] Failures:
[ERROR]   SoapProfileSPITest.ACPAuthContext:744
[ERROR]   SoapProfileSPITest.GetConfigProvider:410 GetConfigProvider failed : AuthConfigFactory.getConfigProvider not called
[ERROR]   SoapProfileSPITest.GetFactory:442 GetFactory failed : AuthConfigFactory.getFactory not called
[ERROR]   SoapProfileSPITest.MessageInfo:597 MessageInfo failed : The request and response messages contains incorrect values
[ERROR]   SoapProfileSPITest.OperationId:713
[ERROR]   SoapProfileSPITest.SecureResponse:527 SecureResponse failed : ServerAuthContext.secureResponse not called
[ERROR]   SoapProfileSPITest.ServerAppContextId:847
[ERROR]   SoapProfileSPITest.ServerAuthConfig:472 ServerAuthConfig failed : AuthConfigProvider.getServerAuthConfig not called
[ERROR]   SoapProfileSPITest.ServerAuthContext:554 TSServerAuthConfig.getAuthContext:  layer=SOAP : appContext=localhost /Hello_web/Hello
[ERROR]   SoapProfileSPITest.ServerRuntimeCallbackSupport:676
[ERROR]   SoapProfileSPITest.ServerRuntimeCommonCallbackSupport:638
[ERROR]   SoapProfileSPITest.ValidateRequest:501 ValidateRequest failed : ServerAuthContext.validateRequest not called
```
The server log again shows no obvious errors.

Counterpart to https://github.com/payara/EngineeringJenkinsjobs/pull/180

### CODE REVIEW

File: authentication-tck/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+```
+mvn clean verify -Ppayara-server-managed -Dpayara.version=${payaraVersion}
+```
+
+### Remote Server Profile
+
+To execute the full TCK against a remote Payara Server installation, start your Payara Server and run the follow asadmin command against it, replacing ${tckDir} and ${payaraHome}:
+
+```
+create-system-properties j2eelogin.name=j2ee:j2eelogin.password=j2ee:provider.configuration.file=${tckDir}/spi/common/ProviderConfiguration.xml:vendor.authconfig.factory=com.sun.jaspic.config.factory.AuthConfigFileFactory:log.file.location=${payaraHome}/glassfish/domains/domain1/logs
```
Comment (breakponchito) [inline authentication-tck/README.md:25]: Now the provider come from the epicyro implementation
```suggestion
create-system-properties j2eelogin.name=j2ee:j2eelogin.password=j2ee:provider.configuration.file=${tckDir}/spi/common/ProviderConfiguration.xml:vendor.authconfig.factory=org.glassfish.epicyro.config.factory.file.AuthConfigFileFactory:log.file.location=${payaraHome}/glassfish/domains/domain1/logs                                    
```

File: authentication-tck/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+```
+create-system-properties j2eelogin.name=j2ee:j2eelogin.password=j2ee:provider.configuration.file=${tckDir}/spi/common/ProviderConfiguration.xml:vendor.authconfig.factory=com.sun.jaspic.config.factory.AuthConfigFileFactory:log.file.location=${payaraHome}/glassfish/domains/domain1/logs                                    
+```
+
+${tckDir} is the directory where the unpacked TCK exists, for example: /home/user/Git/JakartaEE-10-TCK-Runners/authentication-tck/target/authentication-tck-3.1.0/tck
+
+NOTE: On Windows you will need to escape the ':' and '\' characters that will be present on your path variables, due to how the `create-system-properties` parses these characters
+For example:
+```
+create-system-properties "j2eelogin.name=j2ee:j2eelogin.password=j2ee:provider.configuration.file=D\:\\Git\\JakartaEE10-TCK-Runners\\authentication-tck\\target\\authentication-tck-3.1.0\\tck\\spi\\common\\ProviderConfiguration.xml:vendor.authconfig.factory=com.sun.jaspic.config.factory.AuthConfigFileFactory:log.file.location=D\:\\Git\\Payara\\appserver\\distributions\\payara\\target\\stage\\payara7\\glassfish\\domains\\domain1\\logs"
```
Comment (breakponchito) [inline authentication-tck/README.md:33]: same as before:
```suggestion
create-system-properties "j2eelogin.name=j2ee:j2eelogin.password=j2ee:provider.configuration.file=D\:\\Git\\JakartaEE10-TCK-Runners\\authentication-tck\\target\\authentication-tck-3.1.0\\tck\\spi\\common\\ProviderConfiguration.xml:vendor.authconfig.factory=org.glassfish.epicyro.config.factory.fileAuthConfigFileFactory:log.file.location=D\:\\Git\\Payara\\appserver\\distributions\\payara\\target\\stage\\payara7\\glassfish\\domains\\domain1\\logs"
```

**Comment by breakponchito** [comment]
I suspect that probably we need to provided epicyro as a dependency to run the TCK,

**Review by breakponchito — APPROVED** [review]
LGTM, just a few comments but those are not blockers

**Comment by Pandrex247** [comment]
Comments addressed in follow-up PR

**Pull request merged**